README.md

```diff
@@ -30,26 +31,143 @@ From graph to a tree:
+  - we pass the reaction to the second one parameter
+
+### Take API
+`take` is an helper that take two arguments (`take(test, reaction)`):
+ - first argument is the `test`, it can be
+   * **a string :** then the fullpath equality is tested
+   * **a regular expression:** then the fullpath is tested over the regular expression
+   * **a function:** then the function is called and should return `true` so the reaction is called
+     - the function takes two arguments, the `action` to test and the current `tree` (your store instance)
+ - second argument is the `reaction`, this is **a function** that take two arguments (`reaction(action, tree)`):
+   * `action` is the action that pass the test (first argument of `take`)
+   * `tree` is your current store instance, so you can call action on it !
+
+### Action API
+As you can see, the `action` object is given to your `dispatch` function, and to first and second parameters of `take` helper.
+This `action` owns this fields :
+ - **path :** the action path from the root store
+ - **name :** the action name
+ - **fullpath :** `path + '/' + name`
+ - **ended :** for asynchronous action **only**, it means the aynchronous action is ended
+
+## Examples
+We will write 4 ways of doing a router redirection after our login is successful :
+ - `dispatch` is a function (that doesn't return an array)
+ - `dispatch` is a function that returns an array
+   * with a not pure `take` helper function use
+   * with a pure `take` helper function use
+ - `dispatch` is an array
+
+### `dispatch` is a function (that doesn't return an array)
+```es6
+import { addMiddleware } from 'mobx-state-tree'
+import onAction from 'trampss-mst-onaction'
+import Store from './your-store-model'
+
+const store = Store.create({})
+
 const dispatch = (action, tree) => {
   const { fullpath, ended } = action
 
-  if (fullpath === '/auth/login' && ended) {
+  if (ended && fullpath === '/auth/login') {
     tree.ui.router.goToList()
   }
 }
 
-// attach the onAction middleware from trampss-mst-onaction
+addMiddleware(store, onAction(dispatch))
+```
+
+### `dispath` is a function that returns an array - impure take
+```es6
+import { addMiddleware } from 'mobx-state-tree'
+import onAction, { take } from 'trampss-mst-onaction'
+import Store from './your-store-model'
+
+const store = Store.create({})
+
+const dispatch = (action, tree) => [
+  take.ended('/auth/login', () => { tree.ui.router.goToList() }),
+]
+
+addMiddleware(store, onAction(dispatch))
+```
+
+### `dispath` is a function that returns an array - **pure** take
+```es6
+import { addMiddleware } from 'mobx-state-tree'
+import onAction, { take } from 'trampss-mst-onaction'
+import Store from './your-store-model'
+
+const store = Store.create({})
+
+const dispatch = () => [
+  take.ended('/auth/login', (action, tree) => { tree.ui.router.goToList() }),
+]
+
+addMiddleware(store, onAction(dispatch))
+```
+
+### `dispatch` is an array ️❤️
```

Comment:
space between text and emoji

Reply:
![image](https://user-images.githubusercontent.com/6649877/32104899-a76f7d6e-bb26-11e7-9c43-ce209abc1fbe.png)

Reply:
I will not  add extra space for a specific render :p